### DISCUSSION TITLE
Can I Upload Multiple files at same time using Taiko attach api ?

### DISCUSSION THREAD

**Discussion by rkreddy227** [getgauge/taiko#1859] [General]
I have a Scenario to Upload multiple files at same time Can I do that using attach API. Need some help here.

Thanks In Advance.
Ram

**Comment by saikrishna321** [comment]
@rkreddy227 Nope. You need to call the attach Api in loop

**Reply by rkreddy227** [reply]
You mean I need to upload File one by one Correct ? But I think that is not equal to testing Selecting multiple files and uploading them in one go. Can we enhance attach api to accept all files under a folder?

**Comment by NivedhaSenthil** [comment]
This feature is available in the new release of Taiko 1.2.2

**Reply by rkreddy227** [reply]
Awesome I will get the latest version and give it a try Thank you so much or Quick solution

**Comment by rkreddy227** [comment]
I tried to Test the multiple file upload , but i see Single file Upload is also Failing now , I am getting this error. i am using Taiko 1.2.2. 

I beleive File upload attach api is broken in this Release

  Error Message: Error: FileField[id="file-input"]is not visible
            Stacktrace:
            Error: FileField[id="file-input"]is not visible
                at node_modules\taiko\lib\actions\pageActionChecks.js:189:11
                at async waitAndGetActionableElement (node_modules\taiko\lib\actions\pageActionChecks.js:156:3)
                at async attach (node_modules\taiko\lib\actions\attach.js:31:19)
                at async Object.module.exports.attach (node_modules\taiko\lib\taiko.js:1117:16)
                at async Object.module.exports.<computed> [as attach] (node_modules\taiko\lib\taiko.js:2601:14)

**Reply by NivedhaSenthil** [reply]
Please refer if this https://github.com/getgauge/taiko/issues/1771 helps

**Comment by rkreddy227** [comment]
Thank you that worked for me , I was able to read files under a folder and Upload through attach API Now .

Solution 1# Uploading all files under a directory 
const directory = 'resourcesDirectpry/';
   var fileList = fs.readdirSync(directory);
  
   fileList = fileList.map(i => 'resourcesDirectpry/' + i);
       file2.forEach(file => {
           console.log(path.resolve(file));
       });

    await taiko.attach(fileList, taiko.fileField({id: field}),{ force:true });

Solution 2: define files as array an give the array as input to attach api .

 var filearray = ['resources/TestFile.zip', 'resources/TestFile.txt'];
    await taiko.attach(file1, taiko.fileField({id: field}),{ force:true });
